src/primitives/mod.rs

```diff
@@ -96,14 +94,11 @@ impl<E: JubjubEngine> ViewingKey<E> {
         h.update(&preimage);
         let mut h = h.finalize().as_ref().to_vec();
 
-        // Reverse the bytes to interpret it in little-endian byte order
-        h.reverse();
-
-        // Drop the first five bits, so it can be interpreted as a scalar.
-        h[0] &= 0b0000_0111;
+        // Drop the most significant five bits, so it can be interpreted as a scalar.
+        h[31] &= 0b0000_0111;
```

Comment:
Spec says to interpret the BLAKE2s output as a 256-bit integer mod 2<sup>_l_<sub>ivk</sub></sup>, and _l_<sub>ivk</sub> = 251, so this is indeed equivalent to truncating by 5 bits.